File: Agents/dotnet/Shared/Services/HealthCheckService.cs
using System.Collections.Generic;
using System.Net.Http;
using System.Threading.Tasks;

public class HealthCheckService
{
    private readonly HttpClient _httpClient;

    public HealthCheckService(HttpClient httpClient)
    {
        _httpClient = httpClient;
    }

    public async Task<string> GetHealthyEndpointAsync(IEnumerable<string> endpoints, string healthCheckPath = "/health")
    {
        foreach (var endpoint in endpoints)
        {
            if (await IsEndpointHealthyAsync(endpoint, healthCheckPath))
            {
                return endpoint;
            }
        }

        throw new Exception("All endpoints are down.");
    }

    private async Task<bool> IsEndpointHealthyAsync(string endpoint, string healthCheckPath = "/health")
    {
        try
        {
            var response = await _httpClient.GetAsync($"{endpoint}{healthCheckPath}");
            return response.IsSuccessStatusCode;
        }
        catch
        {
            return false;
        }
    }
}

File: Agents/python/LightingAgent/services/health_check_service.py
from flask import Flask, jsonify
import aiohttp
import asyncio

app = Flask(__name__)

class HealthCheckService:
    def __init__(self):
        self.session = aiohttp.ClientSession()

    async def get_healthy_endpoint(self, endpoints, health_check_path="/health"):
        for endpoint in endpoints:
            if await self.is_endpoint_healthy(endpoint, health_check_path):
                return endpoint
        raise Exception("All endpoints are down.")

    async def is_endpoint_healthy(self, endpoint, health_check_path="/health"):
        try:
            async with self.session.get(f"{endpoint}{health_check_path}") as response:
                return response.status == 200
        except:
            return False

    async def close(self):
        await self.session.close()